PlayFabClientSDK/source/PlayFabHttp/PlayFabHttp.cs

```diff
@@ -41,28 +41,15 @@ public class PlayFabJsonSuccess<TResult> where TResult : PlayFabResultCommon
 
     public static class PlayFabHttp
     {
-        private static ITransportPlugin _transport;
-
         public static async Task<object> DoPost(string urlPath, PlayFabRequestCommon request, string authType, string authKey, Dictionary<string, string> extraHeaders)
         {
             if (PlayFabSettings.TitleId == null)
                 throw new Exception("You must set your titleId before making an api call");
-            if (_transport == null)
-                _transport = GetTransport();
-
-            return await _transport.DoPost(urlPath, request, authType, authKey, extraHeaders);
-        }
-
-        private static ITransportPlugin GetTransport()
-        {
-            var transport = (ITransportPlugin)PluginManager.Instance.GetPlugin(PluginContract.Transport);
-            if (transport == null)
-            {
-                transport = new PlayFabTransport();
-                PluginManager.Instance.SetPlugin(PluginContract.Transport, transport);
-            }
-
-            return transport;
+            var transport = (ITransportPlugin)PluginManager.GetPlugin(PluginContract.PlayFab_Transport);
+            if (transport is IPlayFabHttp)
+                return await ((IPlayFabHttp)transport).DoPost(urlPath, request, authType, authKey, extraHeaders);
+            else
+                return await transport.DoPost(urlPath, request, extraHeaders);
```

Comment:
I see what you mean about this being ugly, and this isn't what I was thinking. I think we should ONLY ever do:
transport.DoPost(urlPath, request, extraHeaders);
and never call it this way:
DoPost(urlPath, request, authType, authKey, extraHeaders);
completely remove authType, authKey parameters from transport, and do this instead:
extraHeaders[authType] = authKey;
and then call .DoPost()